docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,214 @@
+
+        channel.hipchat.hipchat_auth_token: '1234556789'
+        channel.hipchat.hipchat_url: 'https://hipchat.foobar.org/'
+
+        # route.group_wait: '30s'
+        # route.repeat_interval: '15m'
+        # route.group_interval: '5m'
+        route.receiver: 'default'
+
+        routes.match.severity: 'email'
+        routes.receiver: 'default'
+        routes.match.severity: 'slack'
+        routes.match.receiver: 'slack_alert'
+
+        receivers.name: 'default'
+        receivers.email_configs.to: 'foo@bar.com'
+
+        receivers.name: 'slack_alert'
+        receivers.slack_configs.channel: '#devops'
+        receivers.slack_configs.send_resolved: 'true'
+
+Iteration 1
+~~~~~~~~~~~
+
+- Define and deploy new namespaced bounded ConfigMap KV stores that will hold
+  user configurations as listed in the requirements
+- Template and render Deployment and Pod manifests that will make use of
+  this user customizable configurations using default pillar values
+- Provide a CLI tool(Bash script) for changing any of the user configurations:
+
+    - Count of replicas for chosen Deployments(Prometheus)
+    - Adding/modifying/removing a Dex authentication connector(OpenLDAP, AD and
+      staticUser store)
+    - Adding/modifying/removing Alertmanager notification configuration
+
+Iteration 2
+~~~~~~~~~~~
+
+- Provide a UI interface that allows CRUD operations on all user customizable
+  settings based on the above requirements
+- Provide a UI interface for adding, updating and deleting service specific
+  configurations for example Dex-LDAP  connector integration.
+- Provide a UI interface for listing MetalK8s available/supported Dex
+  authentication Connectors
+- Provide a UI interface for enabling or disabling Dex authentication
+  connectors(LDAP, Active Directory, StaticUser store)
+- Provide a UI interface for changing the number of replicas for a chosen set
+  of MetalK8s deployments(Prometheus, ....)
+- Add a UI interface for listing Alertmanager notification systems MetalK8s
+  will support(Slack, email, hipchat)
+- Provide a UI interface for adding, modifying and deleting Alertmanager
+  configurations from the listing above
+
+Documentation
+-------------
+
+In the Operational Guide:
+
+* Document how to customize or change any given setting using the CLI tool
+* Document how to customize or change any given setting using the UI interface
+
+Test Plan
+---------
+
+Dex Static User authentication is currently covered in our test-suite and it
+will make sense to cover atleast one other authentication connector with the
+easiest being LDAP since we readily have access to OpenLDAP Docker images and
+automating this process is possible on Scality Cloud.
+
+CRUD Operations on the ConfigMap KV store should be tested both from the UI
+test libraries and the CLI tool provided(An initial idea here is to have a
+simplistic configurator.sh bash script(CLI tool) performing sets of CRUD
+operations on a sample ConfigMap KV store).
```

Comment:
Standard operations to edit configmaps are not super useful to test - what we want to check is propagation of those changes down to the configured services.
E2E tests with CLI and UI are a good thing, because (I guess) those would also trigger the right Salt calls after the CM(s) update(s).

We also need to test edge-cases:
- invalid format for the user configs
- invalid values in the user configs
- lost a user config (someone deleted it)

Reply:
ok